src/components/NumberInput/NumberInput.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import inputIdGenerator from '../../utils/inputIdGenerator';
+
+import './NumberInput.css';
+
+const NumberInput = ({ id = inputIdGenerator.nextIndex, labelText, value, onChange }) => (
+	<div className="NumberInput">
+		<label htmlFor={id}>
+			{labelText}
+		</label>
+		<input
+			id={id}
+			type="number"
+			onChange={e => onChange(e.target.value)}
+			value={value}
+		/>
+	</div>
+);
+
+NumberInput.description = `
+Basic number input complete with label
+`;
+
+NumberInput.propTypes = {
+	/** HTML id attribute for the input
+	If not present one is generated automagically */
+	id: PropTypes.string,
+
+	/** Text used for the label */
+	labelText: PropTypes.string.isRequired,
+
+	/** A function to be called when the value of the input is changed. */
+	onChange: PropTypes.func,
+
+	/** Pre-defined value for input, if present */
+	value: PropTypes.string,
+};
+
+export default NumberInput;
```

Comment:
Isn't this the same as ``<TextInput />`` only with ``type="number"``, is it worth keeping them separate as they generally end up dong the same thing anyway? Or would this be a case of over-atomisation?

Reply:
I've kept this as a <NumberInput /> for the time-being. - Happy to deprecate and remove this later once we have the actual TrustSetting Component. 
Initially I wanted the NumberInput to take in and pass out a `number` but parsing it at each interaction seems a bit much. 

I could go back to the <TextInput /> and add a type param.

 @dan1elhughes @peasandwell - Thoughts?

Reply:
After thinking some more about this I'm happy to keep it as a separate number input component because the number input may have several specific attributes, such as ``pattern``(for touch screen keyboard).